src/game/dust_cloud_brake.c

```diff
@@ -0,0 +1,125 @@
+#include "global.h"
+#include "core.h"
+#include "malloc_vram.h"
+#include "game/game.h"
+
+#include "constants/animations.h"
+
+typedef struct {
+    /* 0x00 */ u16 x;
+    /* 0x02 */ u16 y;
+    /* 0x04 */ u16 unk4;
+    /* 0x06 */ u16 unk6;
+    /* 0x08 */ Sprite s;
+} BrakeDustCloud;
+
+void Task_801F6E0(void);
+void TaskDestructor_801F7A8(struct Task *);
+void Task_801F7B4(void);
+void TaskDestructor_801F7B8(struct Task *);
+
+/* This generates the Dust Clouds that appear while running and then braking by
+ * holding the DPAD in the other direction. */
+
+struct Task *CreateBrakeDustCloud(s32 x, s32 y)
```

Comment:
IMO this should be called `SkidDustCloud` as sonic doesn't have physical brake dust ;)

Reply:
Well, there's the "Skid Attack", so naming it "SkidDustCloud" could be confusing.
https://sonic.fandom.com/wiki/Skid_Attack

I'm sure people will understand that we don't mean Brake Dust from car tires! :D

Reply:
Brake dust specifically refers to the dust from brake pads though :/

Reply:
Does the skid attack cause dust to appear?

Reply:
You could call it ground skid dust?

Reply:
Just double-checked. It's used for both braking and the skid attack.
Maybe "dust_cloud_misc"?

Reply:
Or simply "dust_effect"?
Would change the other to "dust_effect_spindash" in that case.

Reply:
It may also make sense to put them in the same module, since they're directly next to eachother and "do the same thing".

Reply:
I think as there is a clear module split I am ok with them being separate. If the skid attack uses it too, can't it still be called `SkidDustCloud`? I don't see the issue ;)

Reply:
Well, because there's 3 uses of it, and just decalring it to be for the Skid Attack is not accurate, that's my thought process.

(The third one being at the goal, either by manually braking or overshooting the end of the goal.)

Reply:
Isn't the third one also skidding to a stop? That's what I'd call it. It even makes the skidding sound

Reply:
Skid attack
Skid braking
Skidding to a stop

Reply:
Braking dust is better, happy to leave it